ip-utils.c

```diff
@@ -12,13 +12,45 @@
  */
 
 #include <errno.h>
+#include <sys/socket.h>
 #include <sys/types.h>
 #include <arpa/inet.h>
+#include <netdb.h>
 #include <net/if.h>
 #include <string.h>
+#include <unistd.h>
+
+#include <glib.h>
 
 #include "ip-utils.h"
 
+int janus_network_detect_local_ip(char *buf, size_t buflen) {
```

Comment:
The purpose of the ip-utils set of functions is to provide abstractions that makes it easier to write IPv4 & IPv6 compatible code. This function is not IPv6 compatible and does not mark itself as being IPv4. 

Additionally you pass a raw char buffer around for receiving output, which is contrary to conventions for the other functions:
 * It puts the onus on the caller to properly allocate the buffer and pass size (error prone)
 * It is a leaky abstraction, and the 'proper' way to determine the `buflen` creates namespace pollution in the caller context because the correct `buflen` values are should be determined by predefined header constants.
 * This API is inherently hard to use in an IPv4 & IPv6 compatible manner. It is not really fire and forget.

A better (API) would be something like this:
```
   /*
    * addr_type must be one of:
    * - janus_ip_network_query_options_ipv4 => caller wants to support IPv4 only
    * - janus_ip_network_query_options_ipv6 => caller wants to support IPv6 only
    * - janus_ip_network_query_options_any_ip => caller wants to support both IPv4 and IPv6
    */ 
   janus_network_detect_local_ip(janus_ip_network_query_options addr_type, janus_network_address *result);
```

The callee (function) is now free to implement logic for both IPv4 and IPv6 and even better it can reset the `result` to an invalid (null-valued) address which can be tested for in other parts of the code. This allows other parts of the code to recover somewhat from lack of error checking in the caller context, because a null valued result is invalid by definition. So it's more robust.

Then with additional socket/bind/multicast operations also converted to `ip-utils` style the caller could:

 1. Possibly: populate a `struct sockaddr` as follows: `void janus_network_address_to_sockaddr(janus_network_address *addr, int port, struct sockaddr *result)`
 2. Directly open a socket and bind to the address if it wanted to, passing the returned address (if available) and a port number: `int janus_network_address_to_socket(janus_network_address *addr, int port);` and `int janus_network_address_bind_to_socket(janus_network_address *addr, int port);`
 3. Convert to presentation format (strings) via the `janus_network_address_to_string_buffer` call, and then obtain the correct string with: `janus_network_address_string_from_buffer` (already available).